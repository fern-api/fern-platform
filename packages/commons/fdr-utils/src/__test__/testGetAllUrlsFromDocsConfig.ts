import { DocsV2Read } from "@fern-api/fdr-sdk";
import fs from "fs";
import path from "path";
import { getAllUrlsFromDocsConfig } from "../getAllUrlsFromDocsConfig";

export function testGetAllUrlsFromDocsConfig(fixtureName: string): void {
    // eslint-disable-next-line vitest/valid-title
    describe(fixtureName, () => {
        it("gets all urls from docs config", async () => {
            const fixturePath = path.join(__dirname, "fixtures", `${fixtureName}.json`);

            const content = fs.readFileSync(fixturePath, "utf-8");

            const fixture = JSON.parse(content) as DocsV2Read.LoadDocsForUrlResponse;

            const urls = getAllUrlsFromDocsConfig(
                fixture.baseUrl.domain,
                fixture.baseUrl.basePath,
                fixture.definition.config.navigation,
                fixture.definition.apis,
            );

            expect(urls).toMatchSnapshot();
        });
    });
}
