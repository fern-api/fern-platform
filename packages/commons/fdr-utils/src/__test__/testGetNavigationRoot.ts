import { DocsV2Read } from "@fern-api/fdr-sdk";
import fs from "fs";
import path from "path";
import { getNavigationRoot } from "../getNavigationRoot";

export function testGetNavigationRoot(fixtureName: string, slug: string): void {
    // eslint-disable-next-line vitest/valid-title
    describe(fixtureName, () => {
        it("gets navigation root for /" + slug, async () => {
            const fixturePath = path.join(__dirname, "fixtures", `${fixtureName}.json`);

            const content = fs.readFileSync(fixturePath, "utf-8");

            const fixture = JSON.parse(content) as DocsV2Read.LoadDocsForUrlResponse;

            const urls = getNavigationRoot(
                slug.split("/"),
                fixture.baseUrl.basePath,
                fixture.definition.apis,
                fixture.definition.config.navigation,
            );

            expect(stripForSnapshot(urls)).toMatchSnapshot();
        });
    });
}

function stripForSnapshot(obj: ReturnType<typeof getNavigationRoot>): unknown {
    if (obj == null || obj.type === "redirect") {
        return obj;
    }

    // only test the tab index, version index, and current node
    return {
        type: obj.type,
        found: {
            currentTabIndex: obj.found.currentTabIndex,
            currentVersionIndex: obj.found.currentVersionIndex,
            currentNode: obj.found.currentNode,
        },
    };
}
