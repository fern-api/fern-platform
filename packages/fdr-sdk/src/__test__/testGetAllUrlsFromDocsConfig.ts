import urljoin from "url-join";
import { FernNavigation } from "..";
import { NodeCollector } from "../navigation/NodeCollector";

export function testGetAllUrlsFromDocsConfig(
  root: FernNavigation.RootNode,
  domain: string
): void {
  it("gets all urls from docs config", async () => {
    const collector = NodeCollector.collect(root);
    const urls = collector.staticPageSlugs.map((slug) => urljoin(domain, slug));
    expect(urls).toMatchSnapshot();
  });
}
