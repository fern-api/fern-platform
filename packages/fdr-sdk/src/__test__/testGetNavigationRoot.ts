import { FernNavigation } from "..";
import { findNode } from "../navigation/utils";

export function testGetNavigationRoot(root: FernNavigation.RootNode, slug: string): void {
    it("gets navigation root for /" + slug, async () => {
        const found = findNode(root, FernNavigation.slugjoin(slug));

        if (found.type === "found") {
            expect(found.node).toMatchSnapshot();
            expect(found.currentVersion?.versionId).toMatchSnapshot();
            expect(found.currentTab?.slug).toMatchSnapshot();
        } else {
            expect(found).toMatchSnapshot();
        }
    });
}
