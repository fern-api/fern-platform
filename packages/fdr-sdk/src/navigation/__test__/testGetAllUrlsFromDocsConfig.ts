import urljoin from "url-join";
import { FernNavigation } from "../..";
import { NodeCollector } from "../NodeCollector";

export function testGetAllUrlsFromDocsConfig(root: FernNavigation.RootNode, domain: string): void {
    it("gets all urls from docs config", async () => {
        const slugCollector = NodeCollector.collect(root);
        const urls = slugCollector.getPageSlugs().map((slug) => urljoin(domain, slug));
        expect(urls).toMatchSnapshot();
    });
}
