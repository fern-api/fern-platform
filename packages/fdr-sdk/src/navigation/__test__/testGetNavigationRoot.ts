import fs from "fs";
import path from "path";
import { DocsV2Read } from "../../client";
import { convertLoadDocsForUrlResponse, findNode, slugjoin } from "../utils";

export function testGetNavigationRoot(fixtureName: string, slug: string): void {
    // eslint-disable-next-line vitest/valid-title
    describe(fixtureName, () => {
        it("gets navigation root for /" + slug, async () => {
            const fixturePath = path.join(__dirname, "fixtures", `${fixtureName}.json`);

            const content = fs.readFileSync(fixturePath, "utf-8");

            const fixture = JSON.parse(content) as DocsV2Read.LoadDocsForUrlResponse;
            const node = convertLoadDocsForUrlResponse(fixture);
            const found = findNode(node, slugjoin(slug));

            if (found.type === "found") {
                expect(found.node).toMatchSnapshot();
                expect(found.currentVersion?.versionId).toMatchSnapshot();
                expect(found.currentTab?.slug).toMatchSnapshot();
            } else {
                expect(found).toMatchSnapshot();
            }
        });
    });
}
