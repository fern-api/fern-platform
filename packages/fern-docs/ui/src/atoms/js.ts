import type { DocsV1Read } from "@fern-api/fdr-sdk";
import type { JsConfig } from "@fern-docs/ui";
import { isEqual } from "es-toolkit/predicate";
import { selectAtom } from "jotai/utils";
import { DOCS_ATOM } from "./docs";

export const JS_ATOM = selectAtom(DOCS_ATOM, (docs) => docs.js, isEqual);

export function withCustomJavascript(
  readShapeJsConfig: DocsV1Read.JsConfig | undefined,
  resolveFileSrc: (fileId: string) => { src: string } | undefined
): JsConfig | undefined {
  const remote = [
    ...(readShapeJsConfig?.remote ?? []),
    ...(readShapeJsConfig?.files ?? []).map((file) => ({
      url: resolveFileSrc(file.fileId)?.src,
      strategy: file.strategy,
    })),
  ].filter(isRemote);

  const toRet = {
    inline: readShapeJsConfig?.inline,
    remote: remote.length > 0 ? remote : undefined,
  };

  if (!toRet.inline && !toRet.remote) {
    return undefined;
  }

  return toRet;
}

type RemoteJs = NonNullable<JsConfig["remote"]>[number];

function isRemote(remote: {
  url: string | undefined; // potentially undefined if the file is not found
  strategy: RemoteJs["strategy"];
}): remote is RemoteJs {
  return remote.url != null;
}
