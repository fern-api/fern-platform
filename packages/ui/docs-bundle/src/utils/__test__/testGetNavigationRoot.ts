import { DocsV2Read } from "@fern-api/fdr-sdk";
import fs from "fs";
import path from "path";
import { getNavigationRoot } from "../getNavigationRoot";

// eslint-disable-next-line jest/no-export
export function testGetNavigationRoot(fixtureName: string, slug: string): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("gets navigation root for " + slug, async () => {
            const fixturePath = path.join(__dirname, "fixtures", `${fixtureName}.json`);

            const content = fs.readFileSync(fixturePath, "utf-8");

            const fixture = JSON.parse(content) as DocsV2Read.LoadDocsForUrlResponse;

            const urls = getNavigationRoot(
                slug.split("/"),
                fixture.baseUrl.basePath,
                fixture.definition.apis,
                fixture.definition.config.navigation,
            );

            expect(urls).toMatchSnapshot();
        });
    });
}
