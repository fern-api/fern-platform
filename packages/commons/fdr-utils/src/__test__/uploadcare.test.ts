import { testGetAllUrlsFromDocsConfig } from "./testGetAllUrlsFromDocsConfig.js";
import { testGetNavigationRoot } from "./testGetNavigationRoot.js";

const FIXTURE = "uploadcare";

describe("getAllUrlsFromDocsConfig", () => {
    testGetAllUrlsFromDocsConfig(FIXTURE);
});

describe("getNavigationRoot", () => {
    testGetNavigationRoot(FIXTURE, "docs");
    testGetNavigationRoot(FIXTURE, "docs/introduction");
    testGetNavigationRoot(FIXTURE, "docs/introduction/intro");
    testGetNavigationRoot(FIXTURE, "docs/integrations");
    testGetNavigationRoot(FIXTURE, "docs/file-management");
    testGetNavigationRoot(FIXTURE, "docs/file-management/overview");
    testGetNavigationRoot(FIXTURE, "api/url-api");
});
