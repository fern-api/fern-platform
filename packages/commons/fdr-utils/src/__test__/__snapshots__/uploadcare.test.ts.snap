// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAllUrlsFromDocsConfig uploadcare gets all urls from docs config 1`] = `
[
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/introduction",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/introduction/intro",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/introduction/start/quickstart",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/introduction/security",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/introduction/billing",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/uploading",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/uploading/uploads",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/uploading/uploading-files",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/uploading/file-uploader",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/uploading/file-uploader/file-uploader",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/optimization",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/optimization/image-optimization",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/image-transformations",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/image-transformations/overview",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/delivery",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/delivery/overview",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/file-management",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/file-management/overview",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/integrations",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/integrations/all-integrations",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/integrations/api-clients",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/docs/integrations/api-clients/python",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/upload-api",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/upload-api/upload",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/upload-api/upload/base-upload",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/upload-api/upload/multipart-file-upload-start",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/upload-api/upload/multipart-file-upload-part",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/upload-api/upload/multipart-file-upload-complete",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/upload-api/upload/from-url-upload",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/upload-api/upload/from-url-upload-status",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/upload-api/upload/file-upload-info",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/upload-api/groups",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/upload-api/groups/create-files-group",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/upload-api/groups/files-group-info",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file-metadata",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file/files-list",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file/store-file",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file/delete-file-storage",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file/info",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file/files-storing",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file/files-delete",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file/create-local-copy",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file/create-remote-copy",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/add-ons",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/add-ons/aws-rekognition-execute",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/add-ons/aws-rekognition-execution-status",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/add-ons/aws-rekognition-detect-moderation-labels-execute",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/add-ons/aws-rekognition-detect-moderation-labels-execution-status",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/add-ons/uc-clamav-virus-scan-execute",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/add-ons/uc-clamav-virus-scan-execution-status",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/add-ons/remove-bg-execute",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/add-ons/remove-bg-execution-status",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file-metadata/key",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file-metadata/update-file-metadata-key",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/file-metadata/delete-file-metadata-key",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/group",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/group/groups-list",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/group/info",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/group/delete-group",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/project",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/project/info",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/webhook",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/webhook/webhooks-list",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/webhook/create",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/webhook/update-webhook",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/webhook/unsubscribe",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/conversion",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/conversion/document-convert-info",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/conversion/document-convert",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/conversion/document-convert-status",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/conversion/video-convert",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/rest-api-v-0-7/conversion/video-convert-status",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/url-api",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/url-api/file-information",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/url-api/file-information/file-info-json",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/url-api/file-information/file-info-jsonp",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/url-api/file-names",
  "uploadcare-preview-79798ef5-07dc-4ff0-928d-24b4c93ccd4d.docs.buildwithfern.com/api/url-api/file-names/file-with-name",
]
`;

exports[`getNavigationRoot uploadcare gets navigation root for /api/url-api 1`] = `
{
  "redirect": "/api/url-api/file-information/file-info-json",
  "type": "redirect",
}
`;

exports[`getNavigationRoot uploadcare gets navigation root for /docs 1`] = `
{
  "redirect": "/docs/introduction/intro",
  "type": "redirect",
}
`;

exports[`getNavigationRoot uploadcare gets navigation root for /docs/file-management 1`] = `
{
  "redirect": "/docs/file-management/overview",
  "type": "redirect",
}
`;

exports[`getNavigationRoot uploadcare gets navigation root for /docs/file-management/overview 1`] = `
{
  "found": {
    "currentNode": {
      "description": undefined,
      "id": "pages/start/file-management.mdx",
      "slug": [
        "docs",
        "file-management",
        "overview",
      ],
      "title": "Overview",
      "type": "page",
      "urlSlug": "overview",
    },
    "currentTabIndex": 0,
    "currentVersionIndex": -1,
  },
  "type": "found",
}
`;

exports[`getNavigationRoot uploadcare gets navigation root for /docs/integrations 1`] = `
{
  "redirect": "/docs/integrations/all-integrations",
  "type": "redirect",
}
`;

exports[`getNavigationRoot uploadcare gets navigation root for /docs/introduction 1`] = `
{
  "redirect": "/docs/introduction/intro",
  "type": "redirect",
}
`;

exports[`getNavigationRoot uploadcare gets navigation root for /docs/introduction/intro 1`] = `
{
  "found": {
    "currentNode": {
      "description": undefined,
      "id": "pages/index.mdx",
      "slug": [
        "docs",
        "introduction",
        "intro",
      ],
      "title": "Overview",
      "type": "page",
      "urlSlug": "intro",
    },
    "currentTabIndex": 0,
    "currentVersionIndex": -1,
  },
  "type": "found",
}
`;
